### PR TITLE
New test: accepted_range

### PR DESCRIPTION
**Pull request by joellabes**
This is a:
- [ ] bug fix PR with no breaking changes (please change the base branch to `main`)
- [x] new functionality
- [ ] a breaking change

## Description & motivation
This adds an accepted_range test, allowing users to ensure things like that a percentage is between 0 and 100, or that all accounts were created in the past. Technically, if you made a test with a `min_value` of 'A' and `max_value` of '[', I guess you could also check that all names started with a capital letter, but you shouldn't.

I'm open to discussion on the min_value and max_value arguments. I'm tempted to call them greater_than and less_than because I feel like that might make more sense when you're not comparing a scalar number, but I think it's friendlier for the most common case right now.

## Checklist
- [ ] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [x] Redshift
    - [ ] Snowflake
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [ ] I have added an entry to the changelog

### CODE REVIEW

**Comment by clrcrl** [comment]
This looks `:chefs_kiss:`. Lol.

I will test it out later this week :)

File: README.md [hunk 1]
```diff
@@ -418,6 +418,45 @@ An optional `quote_columns` parameter (`default=false`) can also be used if a co
...
+      - name: user_id
+        tests:
+          - dbt_utils.accepted_range:
+              min_value: 0
+              inclusive: false
+      
+      - name: account_created_at
+        tests:
+          - dbt_utils.accepted_range:
+            max_value: "getdate()"
```
Comment (clrcrl) [inline README.md:443]: I think this needs to be indented
```suggestion
              max_value: "getdate()"
```
Same for below snippets

File: README.md [hunk 2]
```diff
@@ -418,6 +418,45 @@ An optional `quote_columns` parameter (`default=false`) can also be used if a co
...
+            #inclusive is true by default
+      
+      - name: num_returned_orders
+        tests:
+          - dbt_utils.accepted_range
```
Comment (clrcrl) [inline README.md:448]: missing a `:`
```suggestion
          - dbt_utils.accepted_range:
```
(below too)

**Comment by joellabes** [comment]
@clrcrl README mistakes are fixed, thanks for catching them! I can see the CI tests failed but can't log into CircleCI to see why - are they mine?

File: integration_tests/models/schema_tests/schema.yml [hunk 3]
```diff
@@ -101,3 +101,20 @@ models:
...
+          
+      - dbt_utils.accepted_range:
+          min_value: -2
+          max_value: 11.1
+          inclusive: false
+          
+      - dbt_utils.accepted_range:
+          min_value: 0
+          inclusive: true
+          where: "id <> -1"
```
Comment (clrcrl) [inline integration_tests/models/schema_tests/schema.yml:120]:
```suggestion
  - name: data_test_accepted_range
    columns:
      - name: id
        tests:
          - dbt_utils.accepted_range:
              min_value: -1
              max_value: 11
              inclusive: true

          - dbt_utils.accepted_range:
              min_value: -2
              max_value: 11.1
              inclusive: false

          - dbt_utils.accepted_range:
              min_value: 0
              inclusive: true
              where: "id <> -1"

```

**Review by clrcrl — CHANGES_REQUESTED** [review]
We're missing a `columns:` entry, which is causing the tests to fail

**Comment by clrcrl** [comment]
Kickin' off those tests now!

**Review by clrcrl — APPROVED** [review]

**Comment by yu-iskw** [comment]
Thank you for the great new feeature. I look forward to using it.
BTW, @clrcrl do you have a timeline to release dbt-utils==0.7.0?

**Comment by clrcrl** [comment]
We'll be doing utils 0.7.0 in line with dbt-core 0.19.0, which I think is coming up in mid-December.

In the meantime, you can either:
(a) install 0.7.0 by including this in your `packages.yml`:

```
packages:
- git: https://github.com/fishtown-analytics/dbt-utils.git
  revision: dev/0.7.0
```

(b) copying this macro into your own project (in the `macros` directory). Note, you won't need to prefix the test name with `dbt_utils` in that case.

FWIW, I'd go with (b), just in case we introduce a breaking change into this branch

**Comment by yu-iskw** [comment]
Thank you for sharing the knowledge. I will try it out.

**Comment by yu-iskw** [comment]
@clrcrl Please let me know when 0.7.0 is out, if you know. If there is any blocker and anything I can contribute to, please let me know as well. Thanks.

**Pull request merged**